homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,172 @@
+"""This component provides support for Reolink binary sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from reolink_aio.api import (
+    FACE_DETECTION_TYPE,
+    PERSON_DETECTION_TYPE,
+    PET_DETECTION_TYPE,
+    VEHICLE_DETECTION_TYPE,
+)
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+from .host import ReolinkHost
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ReolinkBinarySensorDescription(BinarySensorEntityDescription):
+    """A class that describes binary sensor entities."""
+
+    icon: str = "mdi:motion-sensor"
+    icon_off: str = "mdi:motion-sensor-off"
+    value: Callable[[ReolinkHost, int | None], bool | None] = lambda host, ch: None
+    supported: Callable[[ReolinkHost, int | None], bool] = lambda host, ch: True
```

Comment:
Some things I noticed:

- The `value` is always used. Instead of providing a default, we can also make it required.
- Nothing uses `ReolinkHost` direct, all that happens is that the api (`Host`) is grabbed from it. Let's pass that one directly instead, making the lambda a tiny bit smaller and cleaner.
- Channel actually cannot be `None`. This is also incorrectly typed in the constructor of the Reolink base entity.
- The name `Entity` dropped out of `ReolinkBinarySensorEntityDescription` somehow.

```suggestion
@dataclass
class ReolinkBinarySensorEntityDescriptionMixin:
    """Mixin values for Reolink binary sensor entities."""

    value: Callable[[Host, int], bool | None]


@dataclass
class ReolinkBinarySensorEntityDescription(
    BinarySensorEntityDescription, ReolinkBinarySensorEntityDescriptionMixin
):
    """A class that describes binary sensor entities."""

    icon: str = "mdi:motion-sensor"
    icon_off: str = "mdi:motion-sensor-off"
    supported: Callable[[Host, int], bool] = lambda host, ch: True
```